kive/container/models.py

```diff
@@ -921,6 +941,30 @@ def formatted(self):
             text += '*' if self.type == ContainerArgument.INPUT else '/'
         return text
 
+    @property
+    def argtype(self) -> typing.Optional[ContainerArgumentType]:
+        "Classify this argument, or return None if it's unclassifiable."
+        if self.position is not None:
+            if self.allow_multiple:
+                if self.type == self.OUTPUT:
+                    return ContainerArgumentType.FIXED_DIRECTORY_OUTPUT
+            else:
+                if self.type == self.INPUT:
+                    return ContainerArgumentType.FIXED_INPUT
+                elif self.type == self.OUTPUT:
+                    return ContainerArgumentType.FIXED_OUTPUT
+        else:
+            if self.type == self.INPUT:
+                if self.allow_multiple:
+                    return ContainerArgumentType.OPTIONAL_MULTIPLE_INPUT
+                else:
+                    return ContainerArgumentType.OPTIONAL_INPUT
+        # If the above fell through, the model is in a corrupted or partially
```

Comment:
I think an assert would be helpful here.

Reply:
Wait,  does the check in clean() take care of it?

Reply:
I originally threw a `ValidationError` if a type couldn't be assigned, but there are cases (e.g. in tests, dynamically building an argument) were you might want to check if an argtype can be assigned on a partially constructed `ContainerArgument`. I reflected that it might be surprising to have a `ValidationError` thrown when accessing a field, so I added the check to a validation method instead.

This is a less bulletproof check, as the code using `ContainerArgument` has to be using the model validation methods effectively, but I figured it was probably easier to do a pass on the interface code to make sure we're validating these arguments than to review all the code that touches `argtype` to make sure it's never called on a half-initialized instance.

Reply:
In any case, if you think it would be better to have more aggressive state checking, I'm happy to add an exception here.

Reply:
Let's leave it as is, and check it more aggressively if it starts causing pain.